packages/presentation/presentation_publishing/interfaces/unified_search/initialize_time_range.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import fastIsEqual from 'fast-deep-equal';
+import { TimeRange } from '@kbn/es-query';
+import { PublishingSubject } from '../../publishing_subject';
+import { StateComparators } from '../../comparators';
+import { PublishesTimeRange, PublishesWritableTimeRange } from './publishes_unified_search';
+
+export interface SerializedTimeRange {
+  timeRange: TimeRange | undefined;
+}
+
+export const initializeTimeRange = (
```

Comment:
Great to see this following the conventions of the `titles` api. Co-locating the interface, type guards _and_ all of the API logic is such a good win for DX.